krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,238 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+type Interface interface {
+	// GetKubeObject returns the present kubeObject
+	GetKubeObject() *fn.KubeObject
+	//
+	GetGoStruct() (*nephioreqv1alpha1.Interface, error)
+	// GetAttachmentType returns the attachmentType from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetAttachmentType() string
+	// GetCNIType returns the cniType from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetCNIType() string
+	// GetNetworkInstanceName returns the name of the networkInstance from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetNetworkInstanceName() string
+	// SetAttachmentType sets the attachmentType in the spec
+	// returns an error if the value is an unknown type or when the
+	// set fails
+	SetAttachmentType(s string) error
+	// SetCNIType sets the cniType in the spec
+	// returns an error if the value is an unknown type or when the
+	// set fails
+	SetCNIType(s string) error
+	// SetNetworkInstanceName sets the name of the networkInstance in the spec
+	// returns an error when the set fails
+	SetNetworkInstanceName(s string) error
+	// SetSpec sets the spec attributes in the kubeobject
+	SetSpec(*nephioreqv1alpha1.InterfaceSpec) error
+
+	DeleteAttachmentType() error
+
+	DeleteCNIType() error
+}
+
+// NewFromYAML creates a new parser interface
+// It expects a raw byte slice as input representing the serialized yaml file
+func NewFromYAML(b []byte) (Interface, error) {
+	o, err := fn.ParseKubeObject(b)
+	return &itfce{
+		o: o,
+	}, err
+}
+
+// NewFromGoStruct creates a new parser interface
+// It expects a interface go struct
+func NewFromGoStruct(x *nephioreqv1alpha1.Interface) (Interface, error) {
+	b, err := yaml.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+	return NewFromYAML(b)
+}
+
+type itfce struct {
+	o *fn.KubeObject
+}
+
+// GetKubeObject returns the present kubeObject
+func (r *itfce) GetKubeObject() *fn.KubeObject {
+	return r.o
+}
+
+func (r *itfce) GetGoStruct() (*nephioreqv1alpha1.Interface, error) {
```

Comment:
This method would need a test function in `interface_test.go` file

Reply:
added